fabric-resource-loader-v0/src/client/java/net/fabricmc/fabric/mixin/resource/loader/client/PackScreenMixin.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.resource.loader.client;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import net.minecraft.client.gui.screen.pack.PackListWidget;
+import net.minecraft.client.gui.screen.pack.PackScreen;
+import net.minecraft.client.gui.screen.pack.ResourcePackOrganizer;
+
+import net.fabricmc.fabric.impl.client.resource.loader.FabricAbstractPack;
+
+@Mixin(PackScreen.class)
+public class PackScreenMixin {
+	@Inject(method = "method_29672", at = @At("HEAD"), cancellable = true)
+	private void hideHiddenPacks(PackListWidget packListWidget, String string, ResourcePackOrganizer.Pack pack, CallbackInfo ci) {
+		// (Fabric)AbstractPack is the only vanilla implementation of Pack, but we cast anyway.
+		if (pack instanceof FabricAbstractPack fabricAbstractPack && fabricAbstractPack.isHidden()) {
```

Comment:
Maybe interface inject `ResourcePackOrganizer.Pack` instead of `ResourcePackOrganizer.AbstractPack` to avoid this ugly `instanceof` check?

Reply:
Yeah, maybe I should.